### PR TITLE
Do some refactor

### PR DESCRIPTION
**Pull request by mohammadranjbarz**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -163,18 +166,22 @@ class Analytics {
...
+      }
     });
   }
 
   processTrackDataQueue() {
     this.trackQueue.process(numberOfConcurrentJob, async (job, done) => {
       const userPayload = job.data;
-      await this.postUser(userPayload);
-
-      done();
```
Comment (mohammadranjbarz) [inline src/index.ts:177]: in this way if get error and don't call `done` the queue would stuck so we should call `done` even if the job has been failed

File: src/index.ts [hunk 2]
```diff
@@ -87,26 +88,28 @@ class Analytics {
     };
 
     // INSTANCIATE QUEUES
-    this.identifyQueue = new Queue(IDENTIFY_USER_QUEUE, {
+    this.identifyQueue = new Bull<AnalyticsUserPayload>(IDENTIFY_USER_QUEUE, {
```
Comment (mohammadranjbarz) [inline src/index.ts:89]: I think it's better to add types ( with Generic) for queue job data

File: src/index.ts [hunk 3]
```diff
@@ -87,26 +88,28 @@ class Analytics {
...
       redis: this.redisOptions,
       limiter: this.queueRateLimit,
     });
-    this.trackQueue = new Queue(TRACK_DATA_QUEUE, {
+    this.trackQueue = new Bull<AnalyticsDataPayload>(TRACK_DATA_QUEUE, {
       redis: this.redisOptions,
       limiter: this.queueRateLimit,
     });
+    this.processIdentifyUserQueue()
+    this.processTrackDataQueue()
```
Comment (mohammadranjbarz) [inline src/index.ts]: I guess you forgot calling these functions

File: src/index.ts (hunk 1 above)
Comment (CarlosQ96) [reply src/index.ts:177]: Ok ok, got it!

**Review by CarlosQ96 — APPROVED** [review]
Appreciate the review Mohammad! Works good! LGTM merging!

**Pull request merged**